support/utils/command.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"fmt"
+
+	"github.com/vladimirvivien/gexe"
+	"github.com/vladimirvivien/gexe/exec"
+	log "k8s.io/klog/v2"
+)
+
+var commandRunner = gexe.New()
+
+// FindOrInstallGoBasedProvider check if the provider specified by the pPath executable exists or not.
+// If it exists, it returns the path with no error and if not, it uses the `go install` capabilities to
+// install the provider and setup the required binaries to perform the tests. In case if the install
+// is done by this helper, it will return the value for installed binary as provider which can then
+// be set in the in the invoker to make sure the right path is used for the binaries while invoking
+// rest of the workfow after this helper is triggered.
+func FindOrInstallGoBasedProvider(pPath, provider, module, version string) (string, error) {
```

Comment:
Seeing this method sort of remind me of the kubetest2 code. It seems that we are approaching recreating kubetest2 and its provider model. 

It maybe worth taking a look there to see if there is anything we can leverage (if any).  We do have specific needs, but if we can avoid recreating the wheels completely, it's worth checking it out.

Reply:
Yes. The idea is largely inspired by their model of providers. I don't think I found anything on kubetest2 that we can reuse here directly. Do you have anything specific in mind with regards to re-usability ?

Reply:
https://github.com/kubernetes-sigs/kubetest2/blob/e61130e0c75e2dc357d46230993909955a95a646/pkg/types/types.go#L91 

What I found interesting was that, it has defined interfaces much differently. This lets the providers have optional capabilities. But I don't think we need to be that granular in our case here since we want most of the common functionality to be provided by all providers. Using that convention, I have moved the image load helpers out to their own interface so that providers that do not provide that feature can simply ignore that.